src/snapshot/src/persist.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//! Defines an abstract interface for saving/restoring a component from state.
+#![deny(missing_docs)]
+
+/// An abstract interface for saving/restoring a component from state.
```

Comment:
nit: we can replace "from state" with "using its defined state". "from" only makes sense for restoring, and not for saving.

Reply:
Yep, I'll update.